src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -263,32 +257,24 @@ public void ClearMarkers(ScriptFile file)
         /// <param name="settings">The new language server settings.</param>
         public void OnConfigurationUpdated(object sender, LanguageServerSettings settings)
         {
-            InitializeAnalysisEngineToCurrentSettings();
+            if (settings.ScriptAnalysis.Enable ?? true)
+            {
+                InitializeAnalysisEngineToCurrentSettings();
+            }
         }
 
         private void EnsureEngineSettingsCurrent()
         {
-            if (_pssaSettingsFilePath != null
-                && !File.Exists(_pssaSettingsFilePath))
+            if (_analysisEngineLazy == null ||
+                    (_pssaSettingsFilePath != null
+                    && !File.Exists(_pssaSettingsFilePath)))
             {
                 InitializeAnalysisEngineToCurrentSettings();
             }
         }
 
         private void InitializeAnalysisEngineToCurrentSettings()
         {
-            // If script analysis has been disabled, just return null
-            if (_configurationService.CurrentSettings.ScriptAnalysis.Enable != true)
```

Comment:
Did this used to default to false? Seems like we do want to default to true, just want to check that we're changing the behaviour

Reply:
it's default is true https://github.com/PowerShell/PowerShellEditorServices/blob/master/src/PowerShellEditorServices/Services/Workspace/LanguageServerSettings.cs#L71